p2p/simulations/adapters/exec.go

```diff
@@ -107,7 +107,7 @@ func (e *ExecAdapter) NewNode(config *NodeConfig) (Node, error) {
 
 	// listen on a localhost port, which we set when we
 	// initialise NodeConfig (usually a random port)
-	conf.Stack.P2P.ListenAddr = fmt.Sprintf("127.0.0.1:%d", config.Port)
+	conf.Stack.P2P.ListenAddr = fmt.Sprintf(":%d", config.Port)
```

Comment:
`Exec` nodes are now using the preferred outgoing interface, so 127.0.0.1 no longer works. With this change, we listen on all interfaces, which should be fine for tests.